### DISCUSSION TITLE
Error with HAOS v3 Script

### DISCUSSION THREAD

**Discussion by montgomery102** [tteck/Proxmox#297] [Q&A]
First off thank you for all of the scripts, I'm a first time user and looking forward to using some more of them. 

 I'm looking to see if anyone has had the same error as me when running the Home Assistant OS VM script. 
The command I'm running is: `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/vm/haos-vm-v3.sh)"`
The error is: `!!ERROR 255@325 Unknown failure occured.` 
Proxmox version is 7.2-4
I have run it in a SSH sessions as well as the web shell session, using Advanced mode and Default mode always with the same error after Extracted disk Image. Code snip of Advanced run is below. 
```Using Advanced Settings
Using Version 8.2
Using VM ID 5113
Using VM Name ha
Using 2vCPU
Using 4096MiB
Using Bridge vmbr0
Using MAC Address 00:60:2F:6D:1F:58
Using VLAN Tag NONE
Start VM when completed yes
Are these settings correct(y/n)? y

 ✓ Using OMVVMs for Storage Location.
 ✓ Container ID is 5113.
 ✓ Found URL for Home Assistant 8.2 Disk Image
 ✓ https://github.com/home-assistant/operating-system/releases/download/8.2/haos_ova-8.2.qcow2.xz
 ✓ Downloaded 8.2.qcow2 Disk Image
 ✓ Extracted Disk Image
 - Creating HAOS VM...volume 'MYSTORAGE:5113/vm-5113-disk-1.qcow2' does not exist
‼ ERROR 255@325 Unknown failure occured.'```

**Comment by tteck** [comment]
https://github.com/tteck/Proxmox/issues/289
https://github.com/tteck/Proxmox/issues/257

**Comment by tteck** [comment]
Creating a VM without using any media, `lvmthin` is necessary for creation (import disk image).  VM can be moved afterwards.

**Reply by montgomery102** [reply]
Thank you worked as expected!

**Reply by tteck** [reply]
@montgomery102 would you be so kind and try to create a new HAOS VM using OMVVMs for Storage Location?

**Reply by montgomery102** [reply]
@tteck sure can, will take me a little bit to get to it though. Out of curiosity have you made changes, that is where I was trying to create the VM before?

**Reply by tteck** [reply]
Yes, Fixed script :crossed_fingers: to use different storage types.

**Reply by montgomery102** [reply]
Ok. Cool. Ill give it a try in a little bit and let you know.

**Comment by montgomery102** [comment]
Just gave it another try. Got the prompt screen to select storage and selected my shares NFS storage location (OMVVMs). See below for error:
```

        _    _          ____   _____
       | |  | |   /\   / __ \ / ____|
       | |__| |  /  \ | |  | | (___
       |  __  | / /\ \| |  | |\___ \
       | |  | |/ ____ \ |__| |____) |
       |_|  |_/_/ v3 \_\____/|_____/

Using Advanced Settings
Using Version 8.2
Using VM ID 100
Using VM Name haos8.2
Using 2vCPU
Using 4096MiB
Using Bridge vmbr0
Using MAC Address 00:60:2F:37:8A:E1
Using VLAN Tag NONE
Start VM when completed no
Are these settings correct(y/n)? y

 ✓ Using OMVVMs for Storage Location.
 ✓ Container ID is 100.
 ✓ Found URL for Home Assistant 8.2 Disk Image
 ✓ https://github.com/home-assistant/operating-system/releases/download/8.2/haos_ova-8.2.qcow2.xz
 ✓ Downloaded 8.2.qcow2 Disk Image
 ✓ Extracted Disk Image
 - Creating HAOS VM...‼ ERROR 255@324 Unknown failure occured.
```

**Reply by tteck** [reply]
I'm treating `nfs` as a `dir` for storage type. `dir` now is working, but `btrfs` and `nfs` isn't 
```
case $STORAGE_TYPE in
  btrfs|nfs|dir)
        DISK_EXT=".qcow2"
        DISK_REF="$VMID/"
        DIR_IMPORT="-format qcow2"
esac
```
I'll keep trying! Thank you for the help.

**Reply by montgomery102** [reply]
Let me know if you need anything else run, I should be around all day today. I'm east coast USA time zone in case you are somewhere else.

**Reply by montgomery102** [reply]
I just attempted it again with the default setting and it worked successfully. I was looking through my PVE storage and noticed the VM HD from the first attempt was there and thought maybe I'd try again. output of this attempt:
```
Using Default Settings
Using Version 8.2
Using VM ID 100
Using VM Name haos8.2
Using 2vCPU
Using 4096MiB RAM
Using Bridge vmbr0
Using MAC Address 00:60:2F:7E:AA:C5
Using VLAN Tag NONE
Start VM when completed yes
 ✓ Using OMVVMs for Storage Location.
 ✓ Container ID is 100.
 ✓ Found URL for Home Assistant 8.2 Disk Image
 ✓ https://github.com/home-assistant/operating-system/releases/download/8.2/haos_ova-8.2.qcow2.xz
 ✓ Downloaded 8.2.qcow2 Disk Image
 ✓ Extracted Disk Image
 ✓ Created HAOS VM haos8.2
 ✓ Started Home Assistant OS VM
 ✓ Completed Successfully!
```

**Reply by tteck** [reply]
Great news! So, `nfs` storage type IS working.

Thanks!

**Reply by montgomery102** [reply]
Yes!! Thank again for all that you do, and thanks for letting me help. First not monetary contribution for me to a project.
